Feature: new security section, links (#34987)
Expensify/App: README.md

----

# Security
Updated rules for managing all types of chats (add/remove people)

> 1. ### DM

Author: Its not possible on markdown to make ordered list bold unfortunately

----

# Security
Updated rules for managing all types of chats (add/remove people)

1. ### DM
    |  | Member
    | :---: | :---: 
    | **Invite** | ❌ 
    | **Remove** | ❌ 
    | **Leave**  | ❌ 
    | **Can be removed**  | ❌

> - **DM the only ever have 2 participants**

Reviewer: A summary would be good
We can also remove bold style (from all descriptions)
```suggestion
- DM always has two participants. None of the participant can leave or be removed from the DM. Also no additional member can be invited to the chat.
```

2. ### Workspace
    1. #### Workspace
        |   |  Creator  |  Member(Employee/User) | Admin |  Auditor?
        | :---: | :---:  |  :---: | :---: | :---: 
        | **Invite** | ✅ |  ❌ |  ✅ | ❌
        | **Remove** | ✅ |  ❌ |  ✅ | ❌
        | **Leave**  | ❌ |  ✅ |  ❌ | ✅
        | **Can be removed**  | ❌ |  ✅ | ✅ | ✅

        - **Creator can't leave or be removed from their own workspace**
        - **Admins can't leave from the workspace**
        - **Admins can remove other workspace admins, as well as workspace members, and invited guests**
        - **Creator can remove other workspace admins, as well as workspace members, and invited guests**

>         - **Members and Auditoes can't invite or remove from the workspace they were invited to**

Reviewer: ```suggestion
        - **Members and Auditors cannot invite or remove anyone from the workspace**
```

    2. #### Workspace #announce room
        |   |  Member(Employee/User) | Admin |  Auditor?
        | :---: | :---:  |  :---: | :---: 
        | **Invite** | ❌ |  ❌ |  ❌
        | **Remove** | ❌ |  ❌ |  ❌
        | **Leave**  | ❌ |  ❌ |  ❌
        | **Can be removed**  | ❌ |  ❌ |  ❌ |

>        - **All members can't leave or be removed from the #announce room**

Reviewer: ```suggestion
       - **No one can leave or be removed from the #announce room**
```

    3. #### Workspace #admin room
        |   |  Admin |
        | :---: | :---: 
        | **Invite** | ❌  
        | **Remove** | ❌   
        | **Leave**  | ❌ 
        | **Can be removed**  | ❌

        - **Admins can't leave or be removed from #admins**
    
    4. #### Workspace rooms
        |   |  Creator | Member | Guest(outside of the workspace)
        | :---: | :---:  |  :---: | :---:
        | **Invite** | ✅ | ✅ | ✅
        | **Remove** | ✅ | ✅ | ❌
        | **Leave**  | ✅ | ✅ | ✅
        | **Can be removed**  | ✅ | ✅ | ✅

>         - **Everyone can be removed/can leave from the rooms(creator as well)**

Reviewer: ```suggestion
        - **Everyone can be removed/can leave from the room including creator**
```

>         - **Only guests are not able to remove from the room**

Reviewer: ```suggestion
        - **Guests are not able to remove anyone from the room**
```

----

# Security
Updated rules for managing members across all types of chats in New Expensify.

- **Nobody can leave or be removed from something they were automatically added to. For example:**

    - DM members can't leave or be removed from their DMs
    - Members can't leave or be removed from their own workspace chats
    - Admins can't leave or be removed from workspace chats
    - Members can't leave or be removed from the #announce room
    - Admins can't leave or be removed from #admins
    - Domain members can't leave or be removed from their domain chat

>     - Report submitters can't leave or be removed from their reports (eg, if they are the report.accountID)
>     - Report managers can't leave or be removed from their reports (eg, if they are the report.managerID)

Reviewer: Let's remove this. Not really useful here
```suggestion
    - Report submitters can't leave or be removed from their reports
    - Report managers can't leave or be removed from their reports
```

    - Group owners cannot be removed from their groups - they need to transfer ownership first
- **Excepting the above, admins can remove anyone. For example:**
    - Group admins can remove other group admins, as well as group members
    - Workspace admins can remove other workspace admins, as well as workspace members, and invited guests
- **Excepting the above, members can remove guests. For example:**
    - Workspace members can remove non-workspace guests.
- **Excepting the above, anybody can remove themselves from any object**

> 

Reviewer: Let's remove these extra line also
